service/provider/src/main/java/gov/va/vro/service/provider/MasPollingProcessor.java

```diff
@@ -1,42 +1,156 @@
 package gov.va.vro.service.provider;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.va.vro.model.AbdEvidence;
+import gov.va.vro.model.VeteranInfo;
+import gov.va.vro.model.mas.GeneratePdfResp;
 import gov.va.vro.model.mas.MasAutomatedClaimPayload;
+import gov.va.vro.model.mas.MasCollectionAnnotation;
+import gov.va.vro.model.mas.MasCollectionStatus;
+import gov.va.vro.model.mas.MasStatus;
+import gov.va.vro.service.provider.mas.MasException;
+import gov.va.vro.service.provider.mas.service.MasApiService;
+import gov.va.vro.service.provider.mas.service.mapper.MasCollectionAnnotsResults;
+import gov.va.vro.service.spi.model.GeneratePdfPayload;
 import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.camel.Exchange;
 import org.apache.camel.Processor;
 import org.springframework.stereotype.Component;
 
-import java.util.Random;
+import java.util.ArrayList;
+import java.util.List;
 
 @Component
 @RequiredArgsConstructor
 @Slf4j
 public class MasPollingProcessor implements Processor {
-
   private final CamelEntrance camelEntrance;
   private final MasDelays masDelays;
+  private final MasApiService masCollectionAnnotsApiService;
+  private final ObjectMapper objectMapper = new ObjectMapper();
 
   @Override
+  @SneakyThrows
   public void process(Exchange exchange) {
-    var payload = exchange.getMessage().getBody(MasAutomatedClaimPayload.class);
-    log.info("Checking collection status for collection {}.", payload.getCollectionId());
-    // call pcCheckCollectionStatus
-    boolean ready = checkCollectionStatus(payload.getCollectionId());
-    if (ready) {
-      log.info("Collection {} is ready for processing.", payload.getCollectionId());
-      // call pcQueryCollectionAnnots
+
+    // TODO : Remove the extraneous log statements
+    var claimPayload = exchange.getMessage().getBody(MasAutomatedClaimPayload.class);
+
+    boolean isCollectionReady = checkCollectionStatus(claimPayload);
+
+    if (isCollectionReady) {
+      AbdEvidence abdEvidence = getCollectionAnnots(claimPayload);
       // call Lighthouse
       // Combine results and call PDF generation
-      // if a decision is made, call pcOrderExam
+      GeneratePdfResp generatePdfResp = generatePdf(claimPayload, abdEvidence);
+      // call pcOrderExam in the absence of evidence
```

Comment:
I guess it is ok for now but I do not think we want direct calls here in the eventuality. The route for this processor should continue on to the pdf generation on camel. That is the whole reason we are using camel; to do the integration. Same with lighthouse data. We should have in the camel route and have an aggregator of sorts on the route. At least that is how I have always envisioned. For now it is ok if this will be faster.